### PR TITLE
Add mode field into generated vite server build

### PR DESCRIPTION
**Pull request by brophdawg11**
Reported in discord - TS was throwing an error in the vite express template if you use a `server.ts`

### CODE REVIEW

File: packages/remix-dev/vite/plugin.ts [hunk 1]
```diff
@@ -466,6 +467,7 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
         )};`;
       })
       .join("\n")}
+      export const mode = ${JSON.stringify(viteConfig.mode)};
```
[the remark this answers is not in the document]
Comment (pcattori) [reply packages/remix-dev/vite/plugin.ts:470]: Maybe we should have a board for all the things we want to deprecate/remove so we don't forget to do those when we bump a major version

[the remark this answers is not in the document]
Comment (brophdawg11) [reply packages/remix-dev/vite/plugin.ts:470]: yeah I guess we'll want to deprecate after we stabilize vite and officially mark the old compiler as deprecated and remove it in v3?

[the remark this answers is not in the document]
Comment (brophdawg11) [reply packages/remix-dev/vite/plugin.ts:480]: ✅

File: .changeset/tender-cherries-taste.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@remix-run/dev": patch
+---
+
+Add `mode` field into generated serber build for vite
```
Comment (cliffordfajardo) [inline .changeset/tender-cherries-taste.md:5]: `serber` -> `server`

Comment (brophdawg11) [reply .changeset/tender-cherries-taste.md:5]: 🤦  thanks!

```suggestion
Vite: Add `mode` field into generated server build
```

**Pull request merged**